pkg/ee/kubelb/controller.go

```diff
@@ -213,7 +214,7 @@ func (r *reconciler) reconcile(ctx context.Context, cluster *kubermaticv1.Cluste
 }
 
 func (r *reconciler) createOrUpdateKubeLBManagementClusterResources(ctx context.Context, client ctrlruntimeclient.Client, cluster *kubermaticv1.Cluster) error {
-	namespace := cluster.Status.NamespaceName
+	namespace := fmt.Sprintf(kubelbresources.TenantNamespacePattern, cluster.Name)
```

Comment:
It has a 1-1 relation with the user cluster's that have KubeLB enabled; 1 per user cluster.

Reply:
But can't we save all the KubeLB mgmt resources in one namespace and prefix/suffix those resources with the cluster id. Similar how we do for the cluster credentials and how we save them in Kubermatic ns ?

Reply:
Having a dedicated namespace for each "tenant", in this case, user cluster, is ideal for isolating resources such as load balancers and envoy proxy deployments. We have different envoy deployment topologies as well so having dedicated namespaces helps.

I get where you are coming from though but this is how KubeLB is designed; each tenant gets a dedicated namespace and KubeLB CCM is restricted to just that particular namespace. But in terms of KKP since we already take care of that but placing the CCM in seed namespace we don't care about such security implications.

Reply:
Alright, that makes sense. Can this namespace be used with other resources per cluster? I mean something similar to KubeLB in the future ? Do you see that as a valid case ?

Reply:
You are getting it a bit wrong I guess, this namespace is created in the kubeLB management cluster. So in general, no, this should not be used for other resources.